lib/MadNLPGPU/src/MadNLPGPU.jl

```diff
@@ -0,0 +1,14 @@
+module MadNLPGPU
+
+import CUDA: CUBLAS, CUSOLVER, CuVector, CuMatrix, toolkit_version, R_64F
+import MadNLP:
```

Comment:
Currently, the empty linear solver is not declared as a submodule (see [here](https://github.com/sshin23/MadNLP.jl/blob/master/src/LinearSolvers/linearsolvers.jl#L4-L24)). Would there be a benefit of making it a module? We will still need to `import LinearSolvers: introduce, factorize!, solver!, ...`

Reply:
The advantage is mostly in clarifying the code: using `LinearSolvers` as a submodule, we could import only the submodule and then overload directly the different functions. Then, we know directly where are the functions we are overloading (otherwise, we should check all the `import` at the beginning of the file to understand what we are doing exactly)

Reply:
Sure, that makes sense. I'll make the changes